File: bdk-android/lib/src/androidTest/kotlin/org/bitcoindevkit/MnemonicTest.kt
package org.bitcoindevkit

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.runner.RunWith
import kotlin.test.Test
import kotlin.test.assertEquals

@RunWith(AndroidJUnit4::class)
class MnemonicTest {
    // Mnemonics create valid descriptors.
    @Test
    fun mnemonicsCreateValidDescriptors() {
        val mnemonic: Mnemonic = Mnemonic.fromString("space echo position wrist orient erupt relief museum myself grain wisdom tumble")
        val descriptorSecretKey: DescriptorSecretKey = DescriptorSecretKey(Network.TESTNET, mnemonic, null)
        val descriptor: Descriptor = Descriptor.newBip86(descriptorSecretKey, KeychainKind.EXTERNAL, Network.TESTNET)

        assertEquals(
            expected = "tr([be1eec8f/86'/1'/0']tpubDCTtszwSxPx3tATqDrsSyqScPNnUChwQAVAkanuDUCJQESGBbkt68nXXKRDifYSDbeMa2Xg2euKbXaU3YphvGWftDE7ozRKPriT6vAo3xsc/0/*)#m7puekcx",
            actual = descriptor.toString()
        )
    }
}

File: bdk-dart/test/mnemonic_test.dart
import 'package:bdk_dart/bdk.dart';
import 'package:test/test.dart';

import 'test_constants.dart';

void main() {
  group('Mnemonic', () {
    test('produces expected BIP86 descriptor', () {
      final mnemonic = Mnemonic.fromString(
        "space echo position wrist orient erupt relief museum myself grain wisdom tumble",
      );
      final descriptorSecretKey = DescriptorSecretKey(
        Network.testnet,
        mnemonic,
        null,
      );
      final descriptor = Descriptor.newBip86(
        descriptorSecretKey,
        KeychainKind.external_,
        Network.testnet,
      );

      expect(
        descriptor.toString(),
        equals(
          "tr([be1eec8f/86'/1'/0']tpubDCTtszwSxPx3tATqDrsSyqScPNnUChwQAVAkanuDUCJQESGBbkt68nXXKRDifYSDbeMa2Xg2euKbXaU3YphvGWftDE7ozRKPriT6vAo3xsc/0/*)#m7puekcx",
        ),
      );
    });
  });
}
